### PR TITLE
Fixes 500 error if no carrier suitable for order

### PR DESCRIPTION
**Pull request by sudo-do**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Fixes 500 error if no carrier suitable for order.
| Type?             | bug fix
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | 
| Sponsor company   | 
| How to test?      | Remove every carrier in BO, purchase something, a 500 error will shown in purchase step 2. Will be fixed with this PR.

### CODE REVIEW

**Comment by kpodemski** [comment]
Hi @nizelg 

Could you share exact steps to reproduce this error?

File: classes/Hook/HookActionCarrierProcess.php [hunk 1]
```diff
@@ -54,6 +54,10 @@ public function run()
 
             $carrierName = $carrierRepository->findByCarrierId((int) $this->params['cart']->id_carrier);
 
+            if($carrierName == null) {
```
Comment (nicosomb) [inline classes/Hook/HookActionCarrierProcess.php:57]:
```suggestion
            if($carrierName === null) {
```

Comment (nizelg) [reply classes/Hook/HookActionCarrierProcess.php:57]: Exactly

**Review by mflasquin — APPROVED** [review]
LGTM

**Review by mflasquin — COMMENTED** [review]
Seems good, but can you tell us what is the error 500 ? I don't see what can raise a 500 error if $carrierName is null

**Comment by nizelg** [comment]
@mflasquin

https://user-images.githubusercontent.com/104589134/226727651-267afc6c-9383-4344-87e6-6bdb62597469.mp4

Exact error mesage
```
Fatal error: Uncaught TypeError: PrestaShop\Module\Ps_Googleanalytics\Hooks\HookActionCarrierProcess::getGoogleAnalytics4(): Argument #1 ($carrierName) must be of type string, null given, called in /www/wwwroot/hiddenurl.net/modules/ps_googleanalytics/classes/Hook/HookActionCarrierProcess.php on line 60 and defined in /www/wwwroot/hiddenurl.net/modules/ps_googleanalytics/classes/Hook/HookActionCarrierProcess.php:81 Stack trace: #0 /www/wwwroot/hiddenurl.net/modules/ps_googleanalytics/classes/Hook/HookActionCarrierProcess.php(60): PrestaShop\Module\Ps_Googleanalytics\Hooks\HookActionCarrierProcess->getGoogleAnalytics4() #1 /www/wwwroot/hiddenurl.net/modules/ps_googleanalytics/ps_googleanalytics.php(210): PrestaShop\Module\Ps_Googleanalytics\Hooks\HookActionCarrierProcess->run() #2 /www/wwwroot/hiddenurl.net/classes/Hook.php(968): Ps_Googleanalytics->hookActionCarrierProcess() #3 /www/wwwroot/hiddenurl.net/classes/Hook.php(407): HookCore::coreCallHook() #4 /www/wwwroot/hiddenurl.net/classes/Hook.php(903): HookCore::callHookOn() #5 /www/wwwroot/hiddenurl.net/classes/checkout/CheckoutDeliveryStep.php(163): HookCore::exec() #6 /www/wwwroot/hiddenurl.net/classes/checkout/CheckoutProcess.php(76): CheckoutDeliveryStepCore->handleRequest() #7 /www/wwwroot/hiddenurl.net/controllers/front/OrderController.php(252): CheckoutProcessCore->handleRequest() #8 /www/wwwroot/hiddenurl.net/classes/controller/Controller.php(319): OrderControllerCore->initContent() #9 /www/wwwroot/hiddenurl.net/classes/Dispatcher.php(510): ControllerCore->run() #10 /www/wwwroot/hiddenurl.net/index.php(28): DispatcherCore->dispatch() #11 {main} thrown in /www/wwwroot/hiddenurl.net/modules/ps_googleanalytics/classes/Hook/HookActionCarrierProcess.php on line 81
```

**Review by Hlavtox — COMMENTED** [review]
Can you do `if (!empty($carrierName))` instead?

**Comment by Hlavtox** [comment]
@sudo-do Yeah, I think it's alright. There is no better way I think. :-)

**Comment by Hlavtox** [comment]
@sudo-do But keep the early return :P

**Comment by sudo-do** [comment]
@Hlavtox No need to keep `return` or add `else { return; }` because function is over anyway

File: classes/Hook/HookActionCarrierProcess.php [hunk 2]
```diff
@@ -53,13 +53,14 @@ public function run()
...
-                $js = $this->getGoogleAnalytics4($carrierName);
-            } else {
-                $js = $this->getUniversalAnalytics($carrierName);
+            if (!empty($carrierName)) {
+                if ((bool) Configuration::get('GA_V4_ENABLED')) {
+                    $js = $this->getGoogleAnalytics4($carrierName);
+                } else {
+                    $js = $this->getUniversalAnalytics($carrierName);
+                }
+                $ganalyticsDataHandler->manageData($js, 'A');
```
Comment (kpodemski) [inline classes/Hook/HookActionCarrierProcess.php:62]:
```suggestion
            if (empty($carrierName)) {
                return;
            }

            if ((bool) Configuration::get('GA_V4_ENABLED')) {
                $js = $this->getGoogleAnalytics4($carrierName);
            } else {
                $js = $this->getUniversalAnalytics($carrierName);
            }

            $ganalyticsDataHandler->manageData($js, 'A');
```

this way you have an early return and you don't have a nested `if` :) it improves code readability

**Review by Hlavtox — APPROVED** [review]

**Review by MhiriFaten — COMMENTED** [review]
Hello @sudo-do ,

I didn't reproduce the issue as y can see in the attached screenshot below. 
image

Could you please open a new issue and provide the **googleanaytics** version so the issue manager treat it and validate it?

Thank you.

**Comment by sudo-do** [comment]
It's possible that this issue is related to the changes made in this commit: https://github.com/PrestaShop/ps_googleanalytics/pull/139/commits/9b0d38a253ef01ae9fe5ab13a868c673e744c8e8. Unfortunately, I won't be available for about a week, so I won't be able to test it. I have reported the issue to version 4.2.0. If the commit mentioned above is fixing this error in version 4.2.1, this pull request should be merged, as suppressing errors is not recommended.

**Comment by sudo-do** [comment]
Fixed with https://github.com/PrestaShop/ps_googleanalytics/pull/134/files#diff-064b81a68d18df8d995405f980fe7d549fa5a3f5066e90e1a1c2451ee6626b12R85 in release 4.2.1

What do you think @Hlavtox should we merge this or close?

**Comment by kpodemski** [comment]
@sudo-do if it's fixed, it's fixed! thanks for contributing anyway :-)

**Pull request closed without merging**